src/domain/navigation/Navigation.ts

```diff
@@ -60,7 +85,7 @@ export class Navigation {
         this._pathObservable.set(this._path);
     }
 
-    observe(type) {
+    observe(type: keyof SegmentType): SegmentObservable {
```

Comment:
Probably best to make Navigation generic over the SegmentType, so it can be used for different setups (e.g. in apps built on the sdk). EventEmitter is an example of how to do this. The allowsChild function would also be generic over the segment type, so that way you can enforce that the allowsChild function is for the right segment type.